src/DotNetWorker.Core/CollectionModelBindingData.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Functions.Worker.Core
+{
+    /// <summary>
+    /// A representation of Collection of Microsoft.Azure.WebJobs.ParameterBindingData
+    /// </summary>
+    public abstract class CollectionModelBindingData
+    {
+        /// <summary>
+        /// Gets the array of ModelBindingData
+        /// </summary>
+        public abstract ModelBindingData[] ModelBindingDataArray { get; }
```

Comment:
Have we done an API review of these names? Having `Array` at the end of a property name jumped out at me...

Reply:
We had some reviews but might have missed this; I also don't like the look/sound of array here. I also wonder if we need this type at all? Could we not just reuse `GrpcModelBindingData` and have a different constructor for collections? Or something similar?

I.e. instead of `TypedData.DataOneofCase.CollectionModelBindingData => new GrpcCollectionModelBindingData(typedData.CollectionModelBindingData),`

we could do:

`TypedData.DataOneofCase.CollectionModelBindingData => new GrpcModelBindingData(typedData.CollectionModelBindingData),`

cc: @fabiocav @surgupta-msft